packages/suite-base/src/testing/builders/PlayerBuilder.ts

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
+// SPDX-License-Identifier: MPL-2.0
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+//
+// This file incorporates work covered by the following copyright and
+// permission notice:
+//
+//   Copyright 2018-2021 Cruise LLC
+//
+//   This source code is licensed under the Apache License, Version 2.0,
+//   found at http://www.apache.org/licenses/LICENSE-2.0
+//   You may not use this file except in compliance with the License.
+
+import type { MessageEvent } from "@lichtblick/suite";
+import {
+  BlockCache,
+  MessageBlock,
+  PlayerCapabilities,
+  PlayerPresence,
+  PlayerState,
+  PlayerStateActiveData,
+  Progress,
+  SubscribePayload,
+  SubscriptionPreloadType,
+  Topic,
+  TopicSelection,
+  TopicStats,
+} from "@lichtblick/suite-base/players/types";
+import BasicBuilder from "@lichtblick/suite-base/testing/builders/BasicBuilder";
+import RosDatatypesBuilder from "@lichtblick/suite-base/testing/builders/RosDatatypesBuilder";
+import RosTimeBuilder from "@lichtblick/suite-base/testing/builders/RosTimeBuilder";
+import { defaults } from "@lichtblick/suite-base/testing/builders/utilities";
+import { Range } from "@lichtblick/suite-base/util/ranges";
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+class PlayerBuilder {
+  public static subscribePayload(props: Partial<SubscribePayload> = {}): SubscribePayload {
+    return defaults<SubscribePayload>(props, {
+      fields: BasicBuilder.strings(),
+      preloadType: BasicBuilder.sample(["full", "partial"] as SubscriptionPreloadType[]),
```

Comment:
Using generics instead of forcing the type wouldn't be safer? Something like:

BasicBuilder.sample<SubscriptionPreloadType>(["full", "partial"]) ?

I believe it is safer ad the function "sample" already has generics on it.

Just a suggestion.

Reply:
Yes, the problem I faced here is that `SubscriptionPreloadType` is a `type`, if it were an `enum` you could use it just by passing `sample.(SubscriptionPreloadEnum)`. So I had to pass the values ​​and the cast.

Reply:
Actually, the cast is unnecessary because `preloadType ` is waiting for this type.

Reply:
let me try it again

Reply:
done